src/Promise.php

```diff
@@ -94,9 +94,14 @@ public function catch(callable $onRejected): PromiseInterface
         });
     }
 
+    /**
+     * @param callable(): (void|PromiseInterface<void>) $onFulfilledOrRejected
+     * @return PromiseInterface<T>
+     */
     public function finally(callable $onFulfilledOrRejected): PromiseInterface
     {
         return $this->then(static function ($value) use ($onFulfilledOrRejected): PromiseInterface {
+        	/** @var T $value */
```

Comment:
nit: Mixed whitespace, should use spaces only

```suggestion
            /** @var T $value */
```